src/state/psarstake/hederaHooks.ts

```diff
@@ -0,0 +1,337 @@
+
+  const addTransaction = useTransactionAdder();
+  const { t } = useTranslation();
+
+  const png = PNG[chainId];
+
+  const userPngBalance = useTokenBalance(account ?? ZERO_ADDRESS, png);
+
+  // used for max input button
+  const maxAmountInput = maxAmountSpend(chainId, userPngBalance);
+
+  const useUSDCPrice = useUSDCPriceHook[chainId];
+  const usdcPrice = useUSDCPrice(png);
+  const dollerWorth =
+    userPngBalance?.greaterThan('0') && usdcPrice ? Number(typedValue) * Number(usdcPrice.toFixed()) : undefined;
+
+  const wrappedOnDismiss = useCallback(() => {
+    setStakeError(null);
+    setTypedValue('');
+    setStepIndex(0);
+    setHash(null);
+    setAttempting(false);
+  }, []);
+
+  const { parsedAmount, error } = useDerivedStakeInfo(typedValue, png, userPngBalance);
+  const [approval, approveCallback] = useHederaApproveCallback(chainId, parsedAmount, sarStakingContract?.address);
+
+  const mixpanel = useMixpanel();
+
+  const onUserInput = useCallback((_typedValue: string) => {
+    setTypedValue(_typedValue);
+  }, []);
+
+  const handleMax = useCallback(() => {
+    maxAmountInput && onUserInput(maxAmountInput.toExact());
+    setStepIndex(4);
+  }, [maxAmountInput, onUserInput]);
+
+  const onChangePercentage = (value: number) => {
+    if (!userPngBalance) {
+      setTypedValue('0');
+      return;
+    }
+    if (value === 100) {
+      setTypedValue((userPngBalance as TokenAmount).toExact());
+    } else if (value === 0) {
+      setTypedValue('0');
+    } else {
+      const newAmount = (userPngBalance as TokenAmount)
+        .multiply(JSBI.BigInt(value))
+        .divide(JSBI.BigInt(100)) as TokenAmount;
+
+      setTypedValue(newAmount.toSignificant(6));
+    }
+  };
+
+  const {
+    associate: associatePNG,
+    hederaAssociated: isAssociatedPNG,
+    isLoading: isLoadingPNG,
+  } = useHederaTokenAssociated(PNG[chainId]);
+
+  const {
+    associate: associate,
+    hederaAssociated: isAssociated,
+    isLoading: isLoading,
+  } = useHederaAddressAssociated(sarNftContract?.address);
+
+  const onStake = async () => {
+    if (!sarStakingContract || !parsedAmount || !account || isLoading || isLoadingPNG || !associatePNG) {
+      return;
+    }
+    setAttempting(true);
+    try {
+      let response: { hash: string } | null;
+      if (!isAssociatedPNG) {
+        await associatePNG();
+      }
+
+      if (!isAssociated) {
+        await associate();
+      }
+
+      if (!isAssociated || !isAssociatedPNG) return;
+
+      if (!positionId) {
+        response = await hederaFn.sarStake({
+          methodName: 'mint',
+          amount: Number(parsedAmount.raw.toString()),
+          chainId: chainId,
+          account: account,
+        });
+      } else {
+        response = await hederaFn.sarStake({
+          methodName: 'stake',
+          amount: Number(parsedAmount.raw.toString()),
+          chainId: chainId,
+          account: account,
+          positionId: positionId.toNumber(),
+        });
```

Comment:
`parsedAmount.raw.toString()` can be very big number so `amount: Number(parsedAmount.raw.toString())` is not good way to convert it into number because Javascript Number has limitation thats why we are not using direct Number anywhere in dex for big amounts. 
you can send amount into string `parsedAmount.raw.toString()`. and then in `sarStake` method you need to do `as any` because hedera sdk has some issue in type but it is supporting string logically.